tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,285 @@
+        /// <param name="ct">Cancellation token for the operation.</param>
+        /// <returns>
+        /// True if a valid README was generated and passed validation checks; 
+        /// false if the README did not pass validation (e.g., due to dead or invalid links).
+        /// </returns>
+        /// <returns>true if a valid readme was generated, false is the readme did not pass validation checks. For instance, dead/invalid links.</returns>
+        async Task<bool> Generate(ChatClient chatClient, string templatePath, Uri serviceDocumentation, string packagePath, string outputPath, string repoPath, CancellationToken ct)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(templatePath);
+            ArgumentNullException.ThrowIfNull(serviceDocumentation);
+            ArgumentException.ThrowIfNullOrEmpty(outputPath);
+            ArgumentException.ThrowIfNullOrEmpty(repoPath);
+
+            var readmeText = await File.ReadAllTextAsync(templatePath, ct);
+
+            var messages = new ChatMessage[]{
+                new SystemChatMessage(
+                    """
+                    We're going to create some READMEs.
+                    The parameters are:
+                    * A URL that contains service documentation, to be used when creating key concepts, an introduction blurb and any other places where conceptual docs are needed
+                    * A package path that we can use to generate documentation links
+                    Here are some more rules to follow:
+                    - Do not touch the following sections, or its subsections: Contributing.
+                    - Do not generate sample code.
+                    - Rules for proper readmes can be found here: https://github.com/Azure/azure-sdk/blob/main/docs/policies/README-TEMPLATE.md
+                    """
+                ),
+                new SystemChatMessage($"The readme template is this: {readmeText} which we fill in with the user parameters, which follow:"),
+                new UserChatMessage($"Service URL: {serviceDocumentation}"),
+                new UserChatMessage($"Package path: {packagePath}")
+            };
+
+            var response = await chatClient.CompleteChatAsync(messages);
+            var generatedReadmeText = Customize(response.Value.Content[0].Text);
+
+            await File.WriteAllTextAsync(outputPath, generatedReadmeText, ct);
+            output.Output($"Readme written to {outputPath}");
+
+            return await IsValid(repoPath: repoPath, readmePath: outputPath, ct: ct);
+        }
+
+        private async Task<bool> IsValid(string repoPath, string readmePath, CancellationToken ct)
+        {
+            var valid = true;
+
+            try
+            {
+                var result = await VerifyLinks(repoPath, readmePath, ct);
+
+                if (result != null)
+                {
+                    output.OutputError($"[FAIL] Verify-Links.ps1 failed: {result}");
```

Comment:
This is perhaps a poorly named class and some missing docs on my part. Output service is more for final response formatting than for intermediate logging. You should use `ILogger/this.logger` here, and only use output service in `HandleCommand()`.

Reply:
Ah, I wondered what the intended difference was there.

We might be better off saying "this is what you return from your function" instead. Your other comment about the different path for MCP makes me think it'd be good to talk about how to unify these - it is a bit confusing.

Reply:
I swapped it so output was only called once (at the end), and logger was used for the other calls.